vnext/Scripts/msbuildeval.ps1

```diff
@@ -0,0 +1,73 @@
+# Evaluate an MSBuild property name
+param(
+    [Parameter(Mandatory=$true)]
```

Comment:
I know that for react-native-windows it is not the case, but large msbuild based codebases do not have .sln files. They have directory traversal files and use custom private tooling to generate .sln files to edit in visual studio and limmit the number of projects. So technically for rnw projects that are part of larger builds this tool would not work...

Again, not a big deal, as our rnw msbuild files are not suited for integration in larger builds at the moment, but just sharing as FYI.

Reply:
We already have explicit checks elsewhere to make sure that a solution is present or we don't build and error out.



> > * Maybe this can be achieved just wrapping around `MSBuild.exe`?
> >   See `Vars` target at https://github.com/microsoft/react-native-windows/blob/main/vnext/Directory.Build.targets#L8
> 
> You have a cool idea here: We could have a target: `<Target Name="ExtractState">` that would write out a json file that we can then parse from js-side.

Yeah, we absolutely could, but then the question is can it be configured to output *all* variables, or do we have to hardcode the variables we want? That may be a faster, more scoped down option for our current use cases.